base-kv/base-kv-store-balance-controller/src/main/java/com/baidu/bifromq/basekv/balance/KVRangeBalanceController.java

```diff
@@ -155,26 +175,25 @@ private void scheduleNow() {
                     String cmdName = commandToRun.getClass().getSimpleName();
                     Sample start = Timer.start();
                     CommandType commandType = commandToRun.type();
-                    commandRunner.run(commandToRun)
+                    runCommand(commandToRun)
                         .whenCompleteAsync((r, e) -> {
+                            scheduling.set(false);
                             CommandMetrics metrics = metricsManager.getCommandMetrics(balancerName, cmdName);
-                            if (r == CommandRunner.Result.Succeed) {
+                            if (Boolean.TRUE.equals(r)) {
```

Comment:
why not simplify the condition to just if(r)? r could be null?

Reply:
like other whenComplete, it
s better to stick to same coding style: check if e != null then handle v

Reply:
fixed